chain/index/README.MD

```diff
@@ -0,0 +1,188 @@
+
+```toml
+[Fevm]
+EthTxHashMappingLifetimeDays = 0
+
+[Events]
+DisableHistoricFilterAPI = false
+DatabasePath = ""
+```
+
+These options are now deprecated and will not have any effect if used in your configuration file. Please use the `ChainIndexer` config options as described above.
+
+---
+
+## Migration Guide
+
+Migrating to the new `ChainIndexer` involves several steps to ensure a smooth transition. Here's a guide to help you through the process:
+
+1. **Backup Existing Index Databases**
+   - Before restarting your Lotus node, create a backup of your existing index databases.
+   - These files are located in the `{$LOTUS_PATH/sqlite}` directory.
+   - While not used in the migration process, these backups are crucial for potential rollbacks.
+
+2. **Remove Old Index Files**
+   - After creating backups, remove the SQLite database files for `MsgIndex`, `EthTxIndex`, and `EventIndex` from the `{$LOTUS_PATH/sqlite}` directory.
+
+3. **Update Configuration**
+   - Modify your Lotus configuration to enable the `ChainIndexer` as described in the `ChainIndexer Config` section above.
+
+4. **Restart Lotus Node**
+   - Restart your Lotus node with the new configuration.
+   - The `ChainIndexer` will begin indexing **real-time chain state changes** immediately.
+
+Once Lotus starts with the `ChainIndexer` enabled, it will begin indexing real-time chain state changes i.e. new incoming tipsets. However, it will not index any historical chain state i.e. any previously existing chain state. To index historical chain state (aka **"backfilling"**), you can use the following tools that we're shipping with the `ChainIndexer`:
+
+### The `ChainValidateIndex` JSON RPC API
+
+The `ChainValidateIndex` JSON RPC API serves a dual purpose: it validates/diagnoses the integrity of the index at a specific epoch (i.e., it ensures consistency between indexed data and actual chain state), while also providing the option to backfill the `ChainIndexer` if it does have data for the specified epoch. 
+
+```go
+// IndexValidation contains detailed information about the validation status of a specific chain epoch.
+type IndexValidation struct {
+	// TipSetKey is the key of the canonical tipset for this epoch.
+	TipSetKey TipSetKey
+	// Height is the epoch height at which the validation is performed.
+	Height uint64
+	// IndexedMessagesCount indicates the number of indexed messages for the canonical tipset at this epoch.
+	IndexedMessagesCount uint64
+	// IndexedEventsCount signifies the number of indexed events for the canonical tipset at this epoch.
+	IndexedEventsCount uint64
+	// Backfilled denotes whether missing data was successfully backfilled during validation.
+	Backfilled bool
+	// IsNullRound indicates if the epoch corresponds to a null round and therefore does not have any indexed messages or events.
+	IsNullRound bool
+}
+
+// ChainValidateIndex validates the integrity of the chain index at a specified epoch and also optionally backfills missing data.
```

Comment:
Same comments above of lets link to the canonical source for this info and ideally discuss this in term of json-rpc rather than Go.